### PR TITLE
add new keyboard - pikatea macropad

### PR DESCRIPTION
**Pull request by JackPikatea**
## Description

Adding new keyboard configuration for Pikatea Macropad.
http://www.pikatea.com

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pikatea/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL
+};
```
Comment (drashna) [inline keyboards/pikatea/keymaps/default/keymap.c:28]:
```suggestion
```

File: keyboards/pikatea/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+void matrix_scan_user(void) {
+
+}
+
+bool led_update_user(led_t led_state) {
+    return true;
+}
+*/
```
Comment (drashna) [inline keyboards/pikatea/keymaps/default/keymap.c:52]:
```suggestion
```

File: keyboards/pikatea/rules.mk [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/pikatea/rules.mk:18]: You may want to enable bootmagic light if there isn't an easily accessible physical reset button.
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```

File: keyboards/pikatea/config.h [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x7069
+#define PRODUCT_ID      0x706d
```
Comment (fauxpark) [inline keyboards/pikatea/config.h:24]:
```suggestion
#define PRODUCT_ID      0x706D
```

File: keyboards/pikatea/keymaps/default/config.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 JackPikatea
```
Comment (fauxpark) [inline keyboards/pikatea/keymaps/default/config.h:1]: This file can be deleted.

File: keyboards/pikatea/pikatea.c [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    return process_record_user(keycode, record);
+}
+
+bool led_update_kb(led_t led_state) {
+    // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+
+    return led_update_user(led_state);
+}
+*/
```
Comment (fauxpark) [inline keyboards/pikatea/pikatea.c:50]:
```suggestion
```

File: keyboards/pikatea/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# pikatea
+
+![pikatea](https://i.imgur.com/INnCEsB.jpg)
+
+A compact macropad designed to mount to the underside of a desk or monitor (1x5)+Knob. keyboard made and sold by Pikatea Keyboards.
+
+* Keyboard Maintainer: Jack Kester (u/JackPikatea)  
+* Hardware Supported: Pikatea Macropad  
+* Hardware Availability: [Pikatea.com](http://pikatea.com/)
```
Comment (fauxpark) [inline keyboards/pikatea/readme.md:9]:
```suggestion
* Keyboard Maintainer: Jack Kester (u/JackPikatea)
* Hardware Supported: Pikatea Macropad
* Hardware Availability: [Pikatea.com](http://pikatea.com/)
```

File: keyboards/pikatea/rules.mk [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
```
Comment (fauxpark) [inline keyboards/pikatea/rules.mk:16]:
```suggestion
#   change yes to no to disable
```

**Pull request closed without merging**